packages/web-react/src/components/FileUploader/FileUploaderAttachment.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { createRef, RefObject, useEffect } from 'react';
+import classNames from 'classnames';
+import { SpiritFileUploaderAttachmentProps } from '../../types';
+import { useStyleProps } from '../../hooks';
+import { useFileUploaderStyleProps } from './useFileUploaderStyleProps';
+import { Icon } from '../Icon';
+
+const FileUploaderAttachment = (props: SpiritFileUploaderAttachmentProps) => {
+  const { label, name, id, onDismiss, buttonLabel = 'Remove', file, ...restProps } = props;
+
+  const { classProps } = useFileUploaderStyleProps();
+  const { styleProps, props: transferProps } = useStyleProps(restProps);
+
+  const itemReference = createRef() as RefObject<HTMLLIElement>;
+
+  const createAttachmentInput = () => {
+    if (itemReference) {
+      const attachmentInputElement = document.createElement('input');
+      const dataContainer = new DataTransfer();
+
+      if (file) {
+        dataContainer.items.add(file);
+      } else {
+        // TODO: create callback Error
+        console.warn('File not found');
+
+        return;
+      }
+
+      attachmentInputElement.setAttribute('type', 'file');
+      attachmentInputElement.setAttribute('name', `${name}[]`);
+      attachmentInputElement.setAttribute('hidden', '');
+      attachmentInputElement.files = dataContainer.files;
+
+      itemReference.current?.appendChild(attachmentInputElement);
+    }
+  };
+
+  const dismissHandler = () => {
+    if (onDismiss) onDismiss(id);
+  };
+
+  useEffect(() => {
+    createAttachmentInput();
+
+    /* We want to call this hook only once */
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <li
+      id={id}
+      ref={itemReference}
+      {...transferProps}
+      {...styleProps}
+      className={classNames(classProps.attachment.root, styleProps.className)}
+    >
+      <Icon name="file" aria-hidden="true" />
+      <span className="text-truncate">{label}</span>
+      <button type="button" className={classProps.attachment.button} onClick={dismissHandler}>
```

Comment:
Can we have this Button as a component, please?

Reply:
done, but I don't understand why at all

Reply:
Thanks. Because we have in code some similar components, for instance, `TooltipCloseButton` - https://github.com/lmc-eu/spirit-design-system/blob/9e09d4b6f9e7da3ddcebfcbdf038485460482050/packages/web-react/src/components/Tooltip/TooltipCloseButton.tsx. So in the future, we can simpler identify these components and refactor them into one `CloseButton`.